test/validation/api/buffer/buffer.c

```diff
@@ -370,6 +376,56 @@ static void buffer_test_pool_max_pools(void)
 		CU_ASSERT(odp_pool_destroy(pool[i]) == 0);
 }
 
+static void buffer_test_pool_alloc_free(void)
+{
+	odp_pool_param_t param;
+
+	memcpy(&param, &default_param, sizeof(odp_pool_param_t));
+	test_pool_alloc_free(&param);
+	set_cache_size(&param);
+	test_pool_alloc_free(&param);
```

Comment:
The default cache size is implementation specific and for example with linux-generic it's 256. So, perhaps it makes sense to add three test cases: min, default, and max.